### ISSUE TITLE
Argument named "disable_escaping" is not expected

### ISSUE THREAD

**Issue by bskrtich** [hashicorp/terraform-provider-vault#1463]
Argument named "disable_escaping" is not expected in vault_database_secret_backend_connection config

### Terraform Version
Terraform v1.1.9
on linux_amd64
+ provider registry.terraform.io/hashicorp/vault v3.6.0

### Affected Resource(s)
vault_database_secret_backend_connection

### Terraform Configuration Files
```hcl
resource "vault_database_secret_backend_connection" "db" {
  backend = "test"
  name    = "test"

  postgresql {
    disable_escaping = false
    username = "test"
    password = "test"
    connection_url = "postgres://test"
  }
}
```

### Debug Output
```
│ Error: Unsupported argument
│ 
│   on _bootstrap.tf line 35, in resource "vault_database_secret_backend_connection" "db":
│   35:     disable_escaping = false
│ 
│ An argument named "disable_escaping" is not expected here.
```

### Expected Behavior
According to the docs this should be a valid argument
https://registry.terraform.io/providers/hashicorp/vault/latest/docs/resources/database_secret_backend_connection#postgresql-configuration-options

### Actual Behavior
Apply normally

### Steps to Reproduce
1. `terraform apply`

### Important Factoids
I think the tests that would catch this are being skipped.

**Comment by benashz** [comment]
@bskrtich your TF config looks right. I wonder if you wouldn't mind providing us with the version of the provider that you encountered this error with? We added this feature in [3.4.0](https://github.com/hashicorp/terraform-provider-vault/blob/main/CHANGELOG.md#340-march-24-2022).

Thanks,

Ben

**Comment by benashz** [comment]
On second thought, after having taken a closer look, it seems there may be an issue with the `postgresql` engine config. We'll take a look at this soon.

**Comment by bskrtich** [comment]
This is with vault provider v3.6.0. I tested it today.

**Comment by benashz** [comment]
Fixed as a part of #1461

**Comment by benashz** [comment]
Hi @bskrtich , It looks like this regression was introduced in `v3.5.0`. It will be fixed in `v3.7.0`. If you don't require any `v3.5.0+` features then you can pin your provider version to `3.4.1`. See https://www.terraform.io/language/providers/requirements#requiring-providers for more details.

Thanks,

Ben

**Comment by bskrtich** [comment]
yup, thanks.
